src/uthash.h

```diff
@@ -323,13 +375,24 @@ do {
     (add)->hh.prev = NULL;                                                       \
     (head) = (add);                                                              \
     HASH_MAKE_TABLE(hh, head);                                                   \
+    if (HASH_NOMEM_OK && !(head)->hh.tbl) {                                      \
+      (head) = 0;                                                                \
+      SET_MEM_FAILED(&(add)->hh, 1);                                             \
+      uthash_mem_failed(add);                                                    \
```

Comment:
`mem_ok` (local var) is used internally as a parameter for macros that don't have access to hh, e.g. `HASH_BLOOM_MAKE`. I can get rid of `mem_ok` by altering those macros and adding the hh as a parameter to those macros instead. Then, the `hh.mem_failed` will be used instead. OK?

`uthash_mem_failed()` is the callback mechanism. You were adamant that one is needed. I agreed that is useful, because folks may use exceptions, or even `longjmp()` (the memory state will not remain corrupt unlike if used from `uthash_fatal`) as a simpler mechanism of dealing with OOMs. Use of `uthash_mem_failed` and `hh.mem_failed` is complementary to each other. If one had to go on the chopping block, however, it will have to be `uthash_mem_failed()`, because there is no universal access to exceptions, or use of `longjmp()`, however checking 'hh.mem_failed` is always available.